rx/testing/marbles.py

```diff
@@ -154,3 +142,256 @@ def stringify(value):
+            )
+
+    return records
+
+
+TestContext = namedtuple('TestContext', 'start, cold, hot, exp')
+
+
+def test_context(timespan=1):
+    """
+    Initialize a :class:`TestScheduler` and return a namedtuple containing the
+    following functions that wrap its methods.
+
+    :func:`cold()`:
+    Parse a marbles string and return a cold observable
+
+    :func:`hot()`:
+    Parse a marbles string and return a hot observable
+
+    :func:`start()`:
+    Start the test scheduler and invoke the create function,
+    subscribe to the resulting sequence, dispose the subscription and
+    return the resulting records
+
+    :func:`exp()`:
+    Parse a marbles string and return a list of records
+
+    Examples:
+        >>> start, cold, hot, exp = test_context()
+
+        >>> context = test_context()
+        >>> context.cold("--a--b--#", error=Exception("foo"))
+
+        >>> e0 = hot("a---^-b---c-|")
+        >>> ex = exp("    --b---c-|")
+        >>> results = start(e1)
+        >>> assert results.messages == ex
+
+    The underlying test scheduler is initialized with the following
+    parameters:
+        - created time = 100
+        - subscribed = 200
+        - disposed = 1000
+
+    **IMPORTANT**: regarding :func:`hot()`, a marble declared as the
+    fisrt character will be skipped by the test scheduler.
+    E.g. `hot("a--b--")` will only emit `b`.
+    """
+
+    scheduler = TestScheduler()
+    created = 100
+    disposed = 1000
+    subscribed = 200
+
+    def start(create: Union[Observable, Callable[[], Observable]]) -> List[Recorded]:
+
+        def default_create():
+            return create
+
+        if isinstance(create, Observable):
+            create_function = default_create
+        else:
+            create_function = create
+
+        mock_observer = scheduler.start(
+            create=create_function,
+            created=created,
+            subscribed=subscribed,
+            disposed=disposed,
+            )
+        return mock_observer.messages
+
+    def expected(string: str, lookup: Dict = None, error: Exception = None) -> List[Recorded]:
+        if string.find('^') >= 0:
+            raise ValueError(
+                'Expected function does not support subscription symbol "^".'
+                'Got "{}"'.format(string))
+
+        return parse(
+            string,
+            timespan=timespan,
+            time_shift=subscribed,
+            lookup=lookup,
+            error=error,
+            )
+
+    def cold(string: str, lookup: Dict = None, error: Exception = None) -> Observable:
+        if string.find('^') >= 0:
+            raise ValueError(
+                'Cold observable does not support subscription symbol "^".'
+                'Got "{}"'.format(string))
+
+        records = parse(
+            string,
+            timespan=timespan,
+            time_shift=0,
+            lookup=lookup,
+            error=error,
+            )
+        return scheduler.create_cold_observable(records)
```

Comment:
Only test scheduler suports `create_cold_observable`. I'm not sure anymore if we limit marbles/cold/hot to only testing? It could be nice to use it elsewhere as well. That's why I rewrote my marble handling to not use `create_cold_observable` and scheduling the notifications itself.